component/discovery/kubernetes_crds/kubernetes_crds.go

```diff
@@ -0,0 +1,272 @@
+package kubernetes_crds
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	commonConfig "github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/prometheus"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/scrape"
+	"github.com/prometheus/prometheus/storage"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	promop "github.com/prometheus-operator/prometheus-operator/pkg/client/informers/externalversions"
+	versioned "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.kubernetes_crds",
```

Comment:
I don't know if this is the right name for the component, since it's less about just dicsovering things and more focused on telemetry collection. Personally I'd go for `prometheus.kubernetes.pod_monitors`.